pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,312 @@
+import numpy as np
+from pyparsing import Word, alphanums, Suppress, Optional, CharsNotIn, Group, nums, ZeroOrMore, OneOrMore, cppStyleComment, Literal
+from pgmpy.models import BayesianModel
+from pgmpy.factors import TabularCPD
+import re
+
+
+class BifReader(object):
+
+    """
+    Base class for reading network file in bif format
+    """
+    def __init__(self, path=None, string=None):
+
+        """
+        Initialisation of BifReader object
+
+        Parameters
+        ----------------
+        path : file or str
+                File of bif data
+        string : str
+                String of bif data
+        Examples
+        -----------------
+        # dog-problem.bif file is present at
+        # http://www.cs.cmu.edu/~javabayes/Examples/DogProblem/dog-problem.bif
+        >>> reader = BifReader("bif_test.bif")
+        """
+        multi_space = OneOrMore(Literal(' ')|'\t'|'\r'|'\f')                                    # An expression for matching multi spaces
+
+        if path:
+            self.network = open(path, 'r').read()                                               
+
+        elif string:
+            self.network = string        
+
+        else:
+            raise ValueError("Must specify either path or string")
+
+        if '"' in self.network:
+            """
+            Replacing quotes by spaces to remove case sensitivity like:
+            "Dog-Problem" and Dog-problem
+            or "true""false" and "true" "false" and true false
+            """
+            self.network = self.network.replace('"', ' ')
+
+        self.network = multi_space.setParseAction(lambda t:' ').transformString(self.network)   # replacing mulitple spaces or tabs by one space
+
+        if '/*' or '//' in self.network:
```

Comment:
@khalibartan I don't think this is doing what you are expecting because this will always return true.
```python
In [7]: if 1 or 2 in [1, 2, 3]:
   ...:     print (True)
   ...:     
True

In [8]: if 1 or 2 in [1, 3]:
   ...:     print (True)
True
```